src/AI/AIUtils.java

```diff
@@ -458,4 +458,28 @@ public int compare(Unit o1, Unit o2)
       return diff;
     }
   }
+
+  /** Return the set of locations with enemies or terrain that `unit` could attack in one turn from `start` */
+  public static Set<XYCoord> findPossibleTargets(GameMap gameMap, Unit unit, XYCoord start)
+  {
+    Set<XYCoord> targetLocs = new HashSet<XYCoord>();
+    boolean allowEndingOnUnits = false; // We can't attack from on top of another unit.
```

Comment:
I might reverse this, since the units you ordinarily can end on are units that could get out of the way.
If you really wanted to go crazy with it, you could check if units are ready to act and thus get out of the way... though that would introduce a dependency on it being the turn of the CO owning the unit in question.

Reply:
Nah, this is a helper utility, not an AI Submodule. It specifically says it's finding what the unit can do _right now_. We could push `allowEndingOnUnits` into the parameter list to provide an affordance for contemplating multi-move strategies, but since this returns only the targets and not the move locations, that seems dubiously useful. I'd prefer to keep it simple and focused.